doc/policer_counter/PolicerCounter-HLD.md

```diff
@@ -0,0 +1,331 @@
+
+### Scope
+
+This document provides general information about Policer counter feature in SONiC.
+
+### Definitions/Abbreviations 
+
+| Definitions/Abbreviation | Description                                |
+|--------------------------|--------------------------------------------|
+| API                      | Application Programmable Interface         |
+| FC                       | Flex Counter                               | 
+| VID                      | SAIRedis Virtual object identifier |
+| SAI                      | Switch Abstraction Interface               |
+
+### Overview 
+
+Policer Counters provides a simple and scalable approach for monitoring and analyzing traffic policing statistics. 
+This design leverages the existing FC infrastructure, to efficiently collect, process, and store policer statistics such as total packets/bytes and color-based (green/yellow/red) packet/byte counts. By offering real-time visibility into traffic patterns and policy enforcement, this feature enhances network management capabilities, facilitates troubleshooting, and enables fine-tuning of network policies. 
+
+### Requirements
+
+- Support for configuring and managing policer counters through the Flex Counter infrastructure.
+- Ability to retrieve policer counter values, including green, yellow, and red packet/byte counters.
+- Integration with the existing SONiC Flex Counter configuration and management interfaces.
+- Compatibility with the SAI for platform-independent implementation.
+- Using the exsiting  ```counterpoll``` CLI command we are adding support for enabling and disabling polling, and changing polling interval, based on user configuration.
+
+### Architecture Design 
+
+No SONiC architecture changes are required as an existing flex counter infrastructure is being used.
+
+### High-Level Design
+
+<p align=center>
+<img src="img/policer-arch.png" alt="Figure 1. Policer Architecture">
+</p>
+
+### SAI
+
+The attributes that are being queried:
+
+| SAI Attribute                | Description 
+|------------------------------|----------------------|
+| SAI_POLICER_STAT_PACKETS  | Get/set packet count | 
+| SAI_POLICER_STAT_ATTR_BYTES   | Get/set byte count   |
+| SAI_POLICER_STAT_GREEN_PACKETS | Get/set green packet count | 
+| SAI_POLICER_STAT_GREEN_BYTES   | Get/set green byte count   |
+| SAI_POLICER_STAT_YELLOW_PACKETS | Get/set yellow packet count | 
+| SAI_POLICER_STAT_YELLOW_BYTES   | Get/set yellow byte count   |
+| SAI_POLICER_STAT_RED_PACKETS | Get/set red packet count | 
+| SAI_POLICER_STAT_RED_BYTES   | Get/set red byte count   |
+
+### Orchagent
+
+A new type of FC has been added to flex_counter/flex_counter_manager.h for its corresponding SAI object, along with a new FC group named "POLICER":
+
+Counter Type:
+```c++
+CounterType::POLICER
+```
+
+PolicerOrch holds a new object of type FlexCounterManager and is initialized with ```StatsMode::READ```
+and a default polling interval of 10 sec and disable by default:

```

Comment:
"with a default polling interval of 10 seconds." 

I would move "and disabled by default" to after the code block below.


```c++
FlexCounterManager m_pc_manager;
```
m_pc_manager is disabled by default.